src/Assignments/ProblemIframe.tsx

```diff
@@ -60,42 +89,46 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
         setRenderedHTML('');
         // srcdoc='' triggers onLoad with setLoading(false) so setLoading(true) isn't effective until now
         setLoading(true); 
-        (async () => {
-            try {
-                let queryString = qs.stringify(_({
-                    workbookId,
-                    readonly
-                }).omitBy(_.isUndefined).value());
-                if (!_.isEmpty(queryString)) {
-                    queryString = `?${queryString}`;
-                }
 
-                let res;
-                if (previewPath || previewProblemSource) {
-                    res = await postPreviewQuestion({
-                        webworkQuestionPath: previewPath,
-                        problemSeed: previewSeed,
-                        problemSource: previewProblemSource,
-                        showHints: previewShowHints,
-                        showSolutions: previewShowSolutions
-                    });
-                } else {
-                    res = await AxiosRequest.get(`/courses/question/${problem.id}${queryString}`);
-                }
-                // TODO: Error handling.
-                setRenderedHTML(res.data.data.rendererData.renderedHTML);
-            } catch (e) {
-                setError(e.message);
-                logger.error('Error posting preview', e, e.message);
-                setLoading(false);
-            }
-        })();
+        fetchHTML();
```

Comment:
Also this would have been a problem before (and/or is not a problem) but is there a reason we have an orphaned async function and then more logic after it?